### ISSUE TITLE
Allow resuming cloud changes created in unrelated context

### ISSUE THREAD

**Issue by TylerLeonhardt** [microsoft/vscode#174383]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 
- OS Version: 

Steps to Reproduce:

1. Show Cloud Changes
1. Click on the button:
<img width="713" alt="image" src="https://user-images.githubusercontent.com/2644648/218824225-0c183a19-9a05-4e89-ae43-fbd4e8860da1.png">
1. spinner appears but no changes are made locally

at least my change doesn't go away.

Logs don't indicate anything interesting
```
2023-02-14 10:07:55.755 [trace] Sending request to server {"url":"https://vscode-sync.trafficmanager.net/v1/resource/editSessions/6fc9a0f3-9295-456f-8231-2545379a8bca","type":"GET","headers":{"X-Execution-Id":"2fdb13d0-3cd5-4133-86ae-139e0c87dda5","Cache-Control":"no-cache","X-Client-Name":"code-insiders","X-Client-Version":"1.76.0-insider","X-Client-Commit":"af4a70d991053dcbf64d9a86c76a02dced81d9c4","X-Account-Type":"github","X-Machine-Session-Id":"443803c0-8aba-4f32-aa0f-15cd86b156bf","X-User-Session-Id":"2169eda5-f8e5-4637-88e7-f111778e8200"}}
2023-02-14 10:07:55.758 [trace] Sending request to server {"url":"https://vscode-sync.trafficmanager.net/v1/resource/editSessions/6fc9a0f3-9295-456f-8231-2545379a8bca","type":"GET","headers":{"X-Execution-Id":"d10b8282-4321-4762-a743-1a20b4c31543","Cache-Control":"no-cache","X-Client-Name":"code-insiders","X-Client-Version":"1.76.0-insider","X-Client-Commit":"af4a70d991053dcbf64d9a86c76a02dced81d9c4","X-Account-Type":"github","X-Machine-Session-Id":"443803c0-8aba-4f32-aa0f-15cd86b156bf","X-User-Session-Id":"2169eda5-f8e5-4637-88e7-f111778e8200"}}
2023-02-14 10:07:56.194 [trace] Request succeeded {"url":"https://vscode-sync.trafficmanager.net/v1/resource/editSessions/6fc9a0f3-9295-456f-8231-2545379a8bca","status":200,"execution-id":"2fdb13d0-3cd5-4133-86ae-139e0c87dda5"}
2023-02-14 10:07:56.716 [trace] Request succeeded {"url":"https://vscode-sync.trafficmanager.net/v1/resource/editSessions/6fc9a0f3-9295-456f-8231-2545379a8bca","status":200,"execution-id":"d10b8282-4321-4762-a743-1a20b4c31543"}
```

**Comment by TylerLeonhardt** [comment]
my shot in the dark guess is that this Change has some changes that I put in a separate PR that got merged in... so the change in `iconHoverDelegate` is already in main... and in my local branch... but the other changes are not.

**Comment by zombiezen** [comment]
I encountered this issue myself because I had branches that had the same name locally, but their upstream branches were set differently. My local workaround was to `git branch --set-upstream-to=origin/foo` to get Cloud Changes to do the right thing, but the only way I figured this out was by looking at logs. It would be nice for Cloud Changes to at the very least show an error message or something when I click "Resume Working Changes" but it intentionally no-ops.
